app-android/app/src/main/java/de/tutao/tutanota/offline/AndroidSqlCipherFacade.kt

```diff
@@ -21,16 +23,36 @@ class AndroidSqlCipherFacade(private val context: Context) : SqlCipherFacade {
 	private val openedDb: SQLiteDatabase
 		get() = db ?: throw OfflineDbClosedError()
 
+	private val listIdLocks: MutableMap<String, CompletableDeferred<Unit>> = ConcurrentHashMap()
+
 	override suspend fun openDb(userId: String, dbKey: DataWrapper) {
 		// db is volatile so we see the latest value but it doesn't mean that we won't try to open/delete it in parallel
-		// so we need syncrhonized.
+		// so we need synchronized.
 		// Wrap the whole method into synchronized to ensure that no other check or modification can happen in between
 		synchronized(this) {
 			if (db != null) {
 				Log.w(TAG, "opening new database while old one is open")
 				closeDbSync()
 			}
 			db = SQLiteDatabase.openOrCreateDatabase(getDbFile(userId).path, dbKey.data, null)
+
+			// We are using the auto_vacuum=incremental mode to allow for a faster vacuum execution
+			// After changing the auto_vacuum mode we need to run "vacuum" once
+			// auto_vacuum mode: 0 (NONE) | 1 (FULL) | 2 (INCREMENTAL)
+			openedDb.query("PRAGMA auto_vacuum").use { cursor ->
+				if (cursor.moveToNext()) {
+					cursor.readRow().firstNotNullOf {
```

Comment:
I will check again, thanks!

Reply:
I checked again and I think `firstNotNullOf` makes sense here. This way we make sure that the entries in the row are not null, readRow() also doesn't let me run just "first" anyway, which also wouldn't  make sense?, because that would mean that the database does not support "auto_vacuum"? Not 100% sure though...